libs/blocks/table/table.js

```diff
@@ -0,0 +1,462 @@
+  const rowsNum = rows.length;
+
+  const firstRow = rows[0];
+  const colsInRow = firstRow.querySelectorAll('.col');
+  const colsInRowNum = colsInRow.length;
+
+  for (let i = colsInRowNum; i > 0; i--) {
+    const cols = table.querySelectorAll(`.col-${i}`);
+    for (let j = rowsNum - 1; j >= 0; j--) {
+      const currentCol = cols[j];
+      if (!currentCol?.innerText && currentCol?.children.length === 0) {
+        currentCol.classList.add('no-borders');
+      } else {
+        currentCol?.classList.add('border-bottom');
+        break;
+      }
+    }
+  }
+}
+
+function handleMouseOut(cols) {
+  cols.forEach((e) => {
+    e.classList.remove('hover');
+    e.classList.remove('no-top-border');
+    e.classList.remove('hover-border-bottom');
+  });
+}
+
+function handleMouseOver(cols, table, colNum, isCollapseTable, lastSectionHead, lastExpandIcon) {
+  handleMouseOut(cols);
+
+  const headingRow = table.querySelector('.row-heading');
+  const colClass = `col-${colNum}`;
+  const isLastRowCollapsed = lastExpandIcon?.getAttribute('aria-expanded') === 'false';
+
+  cols.forEach((e) => {
+    if (e.classList.contains('col-highlight') && e.innerText) {
+      const matchingCols = Array.from(e.classList).filter(
+        (className) => className.startsWith(colClass),
+      );
+      matchingCols.forEach((className) => {
+        const noTopBorderCol = headingRow.querySelector(`.${className}`);
+        noTopBorderCol.classList.add('no-top-border');
+      });
+    }
+
+    if (isCollapseTable && isLastRowCollapsed) {
+      const lastSectionHeadCol = lastSectionHead.querySelector(`.col-${colNum}`);
+      lastSectionHeadCol.classList.add('hover-border-bottom');
+    }
+
+    e.classList.add('hover');
+  });
+}
+
+function handleHovering(table) {
+  const row1 = table.querySelector('.row-1');
+  const colsInRowNum = row1.childElementCount;
+  const isMerch = table.classList.contains('merch');
+  const startValue = isMerch ? 1 : 2;
+  const isCollapseTable = table.classList.contains('collapse');
+  const sectionHeads = table.querySelectorAll('.section-head');
+  const lastSectionHead = sectionHeads[sectionHeads.length - 1];
+  const lastExpandIcon = lastSectionHead.querySelector('.icon.expand');
+
+  for (let i = startValue; i <= colsInRowNum; i++) {
+    const cols = table.querySelectorAll(`.col-${i}`);
+    cols.forEach((e) => {
+      e.addEventListener('mouseover', () => handleMouseOver(cols, table, i, isCollapseTable, lastSectionHead, lastExpandIcon));
+      e.addEventListener('mouseout', () => handleMouseOut(cols));
+    });
+  }
+}
+
+function handleScrollEffect(table, gnavHeight) {
+  const highlightRow = table.querySelector('.row-highlight');
+  const headingRow = table.querySelector('.row-heading');
+
+  if (highlightRow) {
+    highlightRow.style.top = `${gnavHeight}px`;
+    highlightRow.style.borderTop = '1px solid transparent';
+  } else {
+    headingRow.style.borderTop = '1px solid transparent';
```

Comment:
Is there a reason this is added directly in the javascript? For maintainability, it'd be better to use a class and move these to the stylesheet